lib/Ix/DBIC/Result.pm

```diff
@@ -136,6 +167,91 @@ sub ix_add_properties ($class, @pairs) {
   return;
 }
 
+sub add_array_type ($class, $prop, $def) {
+  my $type = $def->{data_type};
+
+  my $ix_type = $IX_TYPE{ $type };
+
+  Carp::confess("Attempt to add property $prop with unknown data_type $type")
+    unless $ix_type && $ix_type->{data_type};
+
+  my $data_type = $ix_type->{db_data_type} // $ix_type->{data_type};
+
+  my $belongs_to = $class->table;
+  $belongs_to =~ s/s$//;
+
+  my $table = $belongs_to . '_ix_at_' . $prop;
+
+  my $class_name = $class;
+  $class_name .= "::Ix::At::\u$prop";
+
+  my $pkg = <<"EOF";
```

Comment:
Surely(??) this could all be done with method calls and not string eval?  Maybe?  Only weird thing is *maybe* you'd need to change package at runtime?